ipa-core/src/report/hybrid.rs

```diff
@@ -224,35 +363,132 @@ impl<BK, V> HybridReport<BK, V>
 where
     BK: SharedValue,
     V: SharedValue,
+    Replicated<BK>: Serializable,
+    Replicated<V>: Serializable,
+    <Replicated<BK> as Serializable>::Size: Add<U16>,
+    <Replicated<V> as Serializable>::Size: Add<U16>,
+    <<Replicated<BK> as Serializable>::Size as Add<<Replicated<BA64> as Serializable>::Size>>:: Output: ArrayLength,
+    <<Replicated<V> as Serializable>::Size as Add<<Replicated<BA64> as Serializable>::Size>>:: Output: ArrayLength,
 {
+    /// # Panics
+    /// If report length does not fit in `u16`.
+    pub fn encrypted_len(&self) -> u16 {
+        match self {
+            HybridReport::Impression(impression_report) => {
+                impression_report.encrypted_len() +1
+            }
+            HybridReport::Conversion(conversion_report) => {
+                conversion_report.encrypted_len() +1
+            }
+        }
+    }
+
+    /// # Errors
+    /// If there is a problem encrypting the report.
+    pub fn delimited_encrypt_to<R: CryptoRng + RngCore, B: BufMut>(
+        &self,
+        key_id: KeyIdentifier,
+        key_registry: &impl PublicKeyRegistry,
+        info: &HybridInfo,
+        rng: &mut R,
+        out: &mut B,
+    ) -> Result<(), InvalidHybridReportError> {
+        match self {
+            HybridReport::Impression(impression_report) => match info {
+                HybridInfo::Impression(impression_info) =>{
+                    out.put_u16_le(self.encrypted_len());
+                    out.put_u8(1u8);
```

Comment:
this is a nit, but I think we are used to impressions encoded as 0 and conversions as 1. Maybe it is worth maintaining that

Reply:
Good catch, I'll change it